eureka2-server/src/main/java/com/netflix/eureka2/server/resolver/EurekaEndpoint.java

```diff
@@ -0,0 +1,101 @@
+package com.netflix.eureka2.server.resolver;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represent Eureka server host, and service ports.
+ *
+ * @author Tomasz Bak
+ */
+public class EurekaEndpoint {
+
+    public enum ServiceType {Registration, Interest, Replication}
+
+    private final String hostname;
+    private final Integer registrationPort;
+    private final Integer interestPort;
+    private final Integer replicationPort;
+
+    protected EurekaEndpoint(String hostname, int registrationPort, int interestPort, int replicationPort) {
+        this.hostname = hostname;
+        this.registrationPort = registrationPort;
+        this.interestPort = interestPort;
+        this.replicationPort = replicationPort;
+    }
+
+    public String getHostname() {
+        return hostname;
+    }
+
+    public Integer getRegistrationPort() {
+        return registrationPort;
+    }
+
+    public Integer getInterestPort() {
+        return interestPort;
+    }
+
+    public Integer getReplicationPort() {
+        return replicationPort;
+    }
+
+    public Integer getPortFor(ServiceType serviceType) {
+        switch (serviceType) {
+            case Registration:
+                return getRegistrationPort();
+            case Interest:
+                return getInterestPort();
+            default: // == Replication
+                return getReplicationPort();
+        }
+    }
+
+    public static EurekaEndpoint writeServerEndpointFrom(String domainName, int registrationPort, int interestPort, int replicationPort) {
+        return new EurekaEndpoint(domainName, registrationPort, interestPort, replicationPort);
+    }
+
+    public static EurekaEndpoint writeServerEndpointFrom(String hostnameAndPorts) {
```

Comment:
be nice to have javadoc here to specify the expected string format for `hostnameAndPorts`